scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/SupportedEllipticCurvesExtension.java

```diff
@@ -80,26 +72,29 @@ protected void addExtensionData(DatagramWriter writer) {
 		writer.write(listLength + (LIST_LENGTH_BITS / Byte.SIZE), LENGTH_BITS);
 		writer.write(listLength, LIST_LENGTH_BITS);
 
-		for (Integer groupId : supportedGroups) {
-			writer.write(groupId, CURVE_BITS);
+		for (SupportedGroup group : supportedGroups) {
+			writer.write(group.getId(), CURVE_BITS);
 		}
 	}
 
 	public static HelloExtension fromExtensionDataReader(DatagramReader extensionDataReader) {
 
-		List<Integer> groupIds = new ArrayList<Integer>();
+		List<SupportedGroup> groups = new ArrayList<>();
 		int listLength = extensionDataReader.read(LIST_LENGTH_BITS);
 		DatagramReader rangeReader = extensionDataReader.createRangeReader(listLength);
 		while (rangeReader.bytesAvailable()) {
 			int id = rangeReader.read(CURVE_BITS);
-			groupIds.add(id);
+			SupportedGroup group = SupportedGroup.fromId(id);
+			if (group != null && group.isUsable()) {
```

Comment:
This is extension deserialization ? 

I didn't expect that we check that group is usable here ?

Reply:
The ids, where scandium doesn't even know the "supported group name", are also skipped here. So for me it didn't make too much difference.

Reply:
The idea behind this comment : 
Here we deserialize, so we should see what is  sent by the peer.
Later we negotiate and so remove the not usable.

Reply:
For ids, unknown by scandium, this will not work. 
The read list is not really used to visualize, and if, with the comment in the first line, it would require to change the enum implementation, otherwise the unknown gets lost. So, my feeling was, it's not worth to spend too much in things, which are anyway not longer recommended.

Reply:
> For ids, unknown by scandium, this will not work.

I understand.

In term of separation of concern, I still thing separating deserialization & negotiation would be better. But it's not a big deal.

Reply:
I changed to include the unused/unsupported, but not the unknown. Otherwise a second list just with the ids would be required. If that causes trouble in the future, we can add such a raw-list. For dtls details I'm more used to use wireshark than the scandium logs. That convinces more, than the own logs :-).